packages/gatsby-remark-images-contentful/src/__tests__/index.js

```diff
@@ -16,18 +16,28 @@ jest.mock(`../utils/`, () => {
 })
 
 jest.mock(`axios`)
+
+let mockedSharpMetadata
 jest.mock(`sharp`, () => () => {
-  return {
-    metadata: jest.fn(() => {
-      return {
-        width: 200,
-        height: 200,
-        density: 75,
-      }
-    }),
-  }
+  return { metadata: mockedSharpMetadata }
 })
 
+const mockSharpSuccess = () => {
+  mockedSharpMetadata = jest.fn(() => {
+    return {
+      width: 200,
+      height: 200,
+      density: 75,
+    }
+  })
+}
+
+const mockSharpFailure = () => {
+  mockedSharpMetadata = jest.fn(() =>
+    Promise.reject(new Error(`invalid image`))
+  )
+}
```

Comment:
This is the proper way to mock :)

```suggestion
const sharp = require('sharp');

const mockSharpSuccess = () => {
  sharp.metadata.mockResolvedValueOnce({
    width: 200,
    height: 200,
    density: 75,
  })
}

const mockSharpFailure = () => {
  sharp.metadata.mockRejectedValue(new Error(`invalid image`)
}
```

Reply:
see comment above